File: src/main/webapp/scripts/app/system/store/store.js
'use strict';

angular.module('bssuiteApp')
    .config(function ($stateProvider) {
        $stateProvider
            .state('store_management', {
                parent: 'system',
                url: '/store_management',
                data: {
                    authorities: ['ROLE_ADMIN'],
                    pageTitle: 'Store management'
                },
                views: {
                    'content@': {
                        templateUrl: 'scripts/app/system/store/stores.html',
                        controller: 'StoreManagementController'
                    }
                },
                resolve: {
                }
            })
            .state('store_management.new', {
                parent: 'system',
                url: '/store_management/new',
                data: {
                    authorities: ['ROLE_ADMIN'],
                    pageTitle: 'New Store'
                },
                views: {
                    'content@': {
                        templateUrl: 'scripts/app/system/store/store-edit.html',
                        controller: 'StoreManagementEditController'
                    }
                },
                resolve: {
                    entity: function () {
                        return {
                            code: null,
                            name: null,
                            address1: null,
                            address2: null,
                            suburb: null,
                            state: null,
                            postcode: null,
                            country: null,
                            phone: null,
                            fax: null,
                            email: null,
                            webUrl: null,
                            inBusinessSince: null,
                            isArchived: null,
                            id: null
                        };
                    }
                }
            })
            .state('store_management.edit', {
                parent: 'system',
                url: '/store_management/{id}',
                data: {
                    authorities: ['ROLE_ADMIN'],
                    pageTitle: 'Edit store'
                },
                views: {
                    'content@': {
                        templateUrl: 'scripts/app/system/store/store-edit.html',
                        controller: 'StoreManagementEditController'
                    }
                },
                resolve: {
                    entity: ['$stateParams', 'Store', function($stateParams, Store) {
                        return Store.get({id : $stateParams.id});
                    }]
                }

            });
    });

File: src/main/java/com/enginemobi/bssuite/domain/Store.java
package com.enginemobi.bssuite.domain;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.enginemobi.bssuite.domain.util.CustomLocalDateSerializer;
import com.enginemobi.bssuite.domain.util.ISO8601LocalDateDeserializer;
import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;
import org.hibernate.annotations.Type;
import org.joda.time.LocalDate;
import org.springframework.data.elasticsearch.annotations.Document;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;

/**
 * A Store.
 */
@Entity
@Table(name = "store")
@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
@Document(indexName="store")
public class Store implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    
    @Column(name = "code")
    private String code;
    
    @Column(name = "name")
    private String name;
    
    @Column(name = "address1")
    private String address1;
    
    @Column(name = "address2")
    private String address2;
    
    @Column(name = "suburb")
    private String suburb;
    
    @Column(name = "state")
    private String state;
    
    @Column(name = "postcode")
    private String postcode;
    
    @Column(name = "country")
    private String country;
    
    @Column(name = "phone")
    private String phone;
    
    @Column(name = "fax")
    private String fax;
    
    @Column(name = "email")
    private String email;
    
    @Column(name = "web_url")
    private String webUrl;
    
    @Type(type = "org.jadira.usertype.dateandtime.joda.PersistentLocalDate")
    @JsonSerialize(using = CustomLocalDateSerializer.class)
    @JsonDeserialize(using = ISO8601LocalDateDeserializer.class)
    @Column(name = "in_business_since")
    private LocalDate inBusinessSince;
    
    @Column(name = "is_archived")
    private Boolean isArchived;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress1() {
        return address1;
    }

    public void setAddress1(String address1) {
        this.address1 = address1;
    }

    public String getAddress2() {
        return address2;
    }

    public void setAddress2(String address2) {
        this.address2 = address2;
    }

    public String getSuburb() {
        return suburb;
    }

    public void setSuburb(String suburb) {
        this.suburb = suburb;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getPostcode() {
        return postcode;
    }

    public void setPostcode(String postcode) {
        this.postcode = postcode;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getFax() {
        return fax;
    }

    public void setFax(String fax) {
        this.fax = fax;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getWebUrl() {
        return webUrl;
    }

    public void setWebUrl(String webUrl) {
        this.webUrl = webUrl;
    }

    public LocalDate getInBusinessSince() {
        return inBusinessSince;
    }

    public void setInBusinessSince(LocalDate inBusinessSince) {
        this.inBusinessSince = inBusinessSince;
    }

    public Boolean getIsArchived() {
        return isArchived;
    }

    public void setIsArchived(Boolean isArchived) {
        this.isArchived = isArchived;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        Store store = (Store) o;

        if ( ! Objects.equals(id, store.id)) return false;

        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id);
    }

    @Override
    public String toString() {
        return "Store{" +
                "id=" + id +
                ", code='" + code + "'" +
                ", name='" + name + "'" +
                ", address1='" + address1 + "'" +
                ", address2='" + address2 + "'" +
                ", suburb='" + suburb + "'" +
                ", state='" + state + "'" +
                ", postcode='" + postcode + "'" +
                ", country='" + country + "'" +
                ", phone='" + phone + "'" +
                ", fax='" + fax + "'" +
                ", email='" + email + "'" +
                ", webUrl='" + webUrl + "'" +
                ", inBusinessSince='" + inBusinessSince + "'" +
                ", isArchived='" + isArchived + "'" +
                '}';
    }
}
